src/app/components/home/home.component.ts

```diff
@@ -117,6 +121,46 @@ class HomeComponent extends PageComponent implements OnInit {
 
     return `${xPos} ${yPos} ${width} ${height}`;
   }
+
+  /**
+   * The home component must have knowledge on all site locations for the overview map
+   */
+  public fetchAllSiteLocations(): Observable<List<Project | Region>> {
+    const settings = this.config.settings;
+
+    let pageNumber = 1;
+    // Get the first page of points. This will also provide information on the max page length to search through
+    let maxPageNumber = 1;
+
+    while (pageNumber <= maxPageNumber) {
+      const filter = { paging: { page: pageNumber } };
+
+      const sites: Observable<Region[] | Project[]> = settings.hideProjects
+        ? this.regionApi.filter(filter)
+        : this.projectApi.filter(filter);
+
+      this.allSites$ = sites.pipe(
+        map((models) => List<Project | Region>(models)),
+        takeUntil(this.unsubscribe)
+      );
+
+      // find the max page number from the metadata
+      this.allSites$
+        .pipe(takeUntil(this.unsubscribe))
+        .subscribe((site) => maxPageNumber = site.toArray()?.shift()?.getMetadata()?.paging?.maxPage);
+      pageNumber++;
+    }
+
+    return this.allSites$;
+  }
+
+  public projects(sites: List<Project | Region>): List<Project> {
+    return sites?.filter((location) => location.kind === "Project") as List<Project>;
+  }
+
+  public regions(sites: List<Project | Region>): List<Region> {
+    return sites?.filter((location) => location.kind === "Region") as List<Region>;
+  }
```

Comment:
some notes:

- just select all sites
- you can disable paging on the request using the `disable_paging` option https://github.com/QutEcoacoustics/baw-server/wiki/API:-Spec#specialized-patterns
- you must not pull back full sized site objects. Use the projection API filters to only pull back the values you need (id, name, lat/long).
    - you'll need a custom service method

Reply:
Also, I think we can strip all this logic and my suggestions out and move it into the component